stl/src/vector_algorithms.cpp

```diff
@@ -2930,10 +2930,375 @@ namespace {
+                while (_Needle_ptr != _Stop) {
+                    if ((*_Needle_ptr & ~_Ty{0xFF}) != 0) {
+                        return false;
+                    }
+
+                    ++_Needle_ptr;
+                }
+
+                return true;
+            }
+        }
+
+        __m256i __vectorcall _Step(const __m256i _Bitmap, const __m256i _Data) noexcept {
+            __m256i _Data_high    = _mm256_srli_epi32(_Data, 5);
+            __m256i _Bitmap_parts = _mm256_permutevar8x32_epi32(_Bitmap, _Data_high);
+            __m256i _Data_low_inv = _mm256_andnot_si256(_Data, _mm256_set1_epi32(0x1F));
+            __m256i _Mask         = _mm256_sllv_epi32(_Bitmap_parts, _Data_low_inv);
+            return _Mask;
+        }
+
+        template <class _Ty>
+        __m256i _Load_avx_256_8(const _Ty* const _Src) noexcept {
+            if constexpr (sizeof(_Ty) == 1) {
+                return _mm256_cvtepu8_epi32(_mm_loadu_si64(_Src));
+            } else if constexpr (sizeof(_Ty) == 2) {
+                return _mm256_cvtepu16_epi32(_mm_loadu_si128(reinterpret_cast<const __m128i*>(_Src)));
+            } else if constexpr (sizeof(_Ty) == 4) {
+                return _mm256_loadu_si256(reinterpret_cast<const __m256i*>(_Src));
+            } else if constexpr (sizeof(_Ty) == 8) {
+                const __m256i _Low  = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(_Src));
+                const __m256i _High = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(_Src) + 1);
+                const __m256i _Pack = _mm256_packs_epi32(_Low, _High);
+                return _mm256_permutex_epi64(_Pack, _MM_SHUFFLE(3, 1, 2, 0));
+            } else {
+                static_assert(false, "Unexpected size");
+            }
+        }
+
+        template <class _Ty>
+        __m256i _Mask_out_oveflow(const __m256i _Mask, const __m256i _Data) noexcept {
+            if constexpr (sizeof(_Ty) == 1) {
+                return _Mask;
+            } else {
+                const __m256i _Data_high = _mm256_and_si256(_Data, _mm256_set1_epi32(static_cast<int>(0xFFFF'FF00)));
+                const __m256i _Fit_mask  = _mm256_cmpeq_epi32(_Data_high, _mm256_setzero_si256());
+                return _mm256_and_si256(_Mask, _Fit_mask);
+            }
+        }
+
+        template <class _Ty>
+        __m256i _Make_bitmap_small(const _Ty* _Needle_ptr, const size_t _Needle_length) noexcept {
+            __m256i _Bitmap = _mm256_setzero_si256();
+
+            const _Ty* const _Stop = _Needle_ptr + _Needle_length;
+
+            for (; _Needle_ptr != _Stop; ++_Needle_ptr) {
+                const _Ty _Val                = *_Needle_ptr;
+                const __m256i _Count_low      = _mm256_broadcastq_epi64(_mm_cvtsi32_si128(_Val & 0x3F));
+                const uint32_t _One_1_high    = 1u << uint32_t((_Val >> 3) & 0x18);
+                const __m256i _One_1_high_unp = _mm256_cvtepu8_epi64(_mm_cvtsi32_si128(_One_1_high));

```

Comment:
"Unpacked", meaning u8[8] to u32[8] unpacking

Reply:
Separated steps differently to avoid having `_One_1_high` vs `_One_1_high_unp`. Now the names should make more sense.

There isn't in fact any deep meaning in particular variables naming and steps separation. Just making the whole thing a single statement would be unreadable, but making each operation a single line would also be harder to comprehend.

Reply:
Hm, agree with that. "unpack" operation is known in x86 SIMD, but the intrinsics use the full word "unpack", and even assembler instructions are "unpck", but not "unp"

> (consider the Standard's `showmanyc`).

👀 show many what? 👀 I'm lucky I never needed it.

Reply:
`showmanyc` has nothing on `xsgetn`!